device/globals/game.gd

```diff
@@ -99,7 +99,20 @@ func set_current_tool(p_tool):
 	current_tool = p_tool
 
 func clicked(obj, pos, input_event = null):
+	if !vm.can_interact():
+		return
+
+	if !player:
+		return
+
+	if mode != "default":
+		return
+
 	var walk_context = null
+	var action = obj.get_action()
+
+	# XXX: Why does this disable joystick_mode?
+	joystick_mode = false
```

Comment:
I think it's because joystick and mouse input mouse don't work well together. Eg. when in joystick mode, the action menu is spawned when the player is within a certain distance of an object, and the code on lines 127 and 128 closes it on every click, which might be inconvenient. Presumably, if the mouse is clicked, it means that the game is not being played with a joystick.

Reply:
This sounds like it requires more than a slight refactor... Which is outside the scope of this PR, IMO.